testing/test_plan.md

```diff
@@ -0,0 +1,135 @@
+# Test plan
+
+## Environment
+
+-   OS: XXX (Windows, macOS, latest Ubuntu LTS)
+-   Python
+    -   Distribution: XXX (CPython, miniconda)
+    -   Version: XXX (3.9, 3.10, 3.11)
+-   VS Code: XXX (Insiders)
+-   Python && Pylance Extension: XXX (Insiders)
+
+## Tests
+
+### P0 Test Scenarios
+
+#### Basic language services
+
+-   [ ] Errors
+    1. Open a new python file
+    1. Type invalid python into it. Like `x2+2-3` on a single line.
+    1. Verify you get a red squiggle
+    1. Type valid python after that. 
+    1. Make sure no squiggles happen
+-   [ ] Import errors
+    1. Create a brand new python environment
+    1. Select this in VS code.
+    1. Open a new python file
+    1. Add `import pandas as pd` at the top
+    1. It should error on the pandas
+    1. Open a terminal
+    1. Activate your new environment
+    1. Install pandas (`python -m pip install pandas`)
+    1. Go back to the python file in VS code
+    1. There should no longer be an error for importing pandas.
+-   [ ] Completions
+    1. Open a new python file
+    1. Pick an interpreter on the bottom right
+    1. Add an import for sys `import sys`
+    1. Type `sys.` and verify you get completions
+-   [ ] Hover
+    1. Open a new python file
+    1. Pick an interpreter on the bottom right if not already selected.
+    1. Add an import for sys `import sys`
+    1. Hover over `sys` and make sure it shows something
+    1. Add a new function, like so
+    ```python
+    def foo():
+        '''
+        Doc string for foo 
+        '''
+        pass
+    ```
+    6. Hover over `foo` and make sure you get a doc string
+    [ ] Goto definition
+    1. Open a new python file
+    1. Add an import for sys `import sys`
+    1. Go to definition on the `sys` module.
+    1. It should jump to the definition of sys in your python environment
+
+#### Notebooks
+
+-   [ ] Imports from files
+    1. Open a new jupyter notebook
+    1. Open a new python file
+    1. Create a method in the python file
+    1. In the first cell of the notebook, import the python file
+    1. Verify no red squiggles show up
+    1. Try calling the method from the python file. Something like so:

```

Comment:
Rephrasing to ensure that "call the method from the python file" is not misinterpreted as "call the method by adding a call in the python file".

```suggestion
    1. In the same cell of the notebook, try calling the method you created. Something like so:

```